packages/react-devtools-shared/src/devtools/views/Components/HookNamesContext.js

```diff
@@ -2,16 +2,19 @@
 
 import {createContext} from 'react';
 import type {
+  FetchFileWithCaching,
   LoadHookNamesFunction,
   PurgeCachedHookNamesMetadata,
 } from '../DevTools';
 
 export type Context = {
+  fetchFileWithCaching: FetchFileWithCaching | null,
```

Comment:
why did you decide to pass this via context and as an arg to the load function as opposed to directly using it?

Reply:
Code in react-devtools-shared can't require code in react-devtools-extensions, only the other way around. Soon (after this PR) we'll move most of the named hooks code into react-devtools-shared and just inject the `fetchFileWithCaching` helper from `main` though. (This is necessary to support named hooks for the react-devtools-inline package.)